yroots/IntervalChecks.py

```diff
@@ -117,24 +156,25 @@ def check_subintervals(self, subintervals, scaled_subintervals, polys, change_si
             The MultiCheb polynomials that approximate the functions on these intervals.
         change_sign: list
             A list of bools of whether we know the functions can change sign on the subintervals.
-        approx_tol: float
-            The sup norm bound on the approximation error.
+        errors: list
+            The approximation errors of the polynomials.
         Returns
         -------
         check_interval : bool
             True if we can throw out the interval. Otherwise False.
         '''
         for check in self.subinterval_checks:
-            for poly in polys:
-                mask = check(poly, scaled_subintervals, change_sign, approx_tol)
+            for poly,error in zip(polys, errors):
+                mask = check(poly, scaled_subintervals, change_sign, error)
                 new_scaled_subintervals = []
                 new_subintervals = []
                 for i, result in enumerate(mask):
                     if result:
                         new_scaled_subintervals.append(scaled_subintervals[i])
                         new_subintervals.append(subintervals[i])
                     else:
-                        self.track_interval(check.__name__, subintervals[i])
+                        if not self.polishing:
+                            self.track_interval(check.__name__, subintervals[i])
```

Comment:
Doesn't track the intervals in polishing, we could add something to do this but I figured to jus track the first run is good enough for now